subprojects/core/src/integTest/groovy/org/gradle/api/file/FileCollectionSymlinkIntegrationTest.groovy

```diff
@@ -223,6 +203,51 @@ class FileCollectionSymlinkIntegrationTest extends AbstractIntegrationSpec {
         run 'inputBrokenLinkNameCollector'
         then:
         skipped ':inputBrokenLinkNameCollector'
+
+        where:
+        type             | createBuildFile
+        'InputDirectory' | taskWithInputFiles
+        'InputFiles'     | taskWithInputDirectory
+    }
+
+    static taskWithInputFiles = { AbstractIntegrationSpec context, TestFile inputFileTarget, TestFile output ->
```

Comment:
Those static closures make the test harder to read for me. Maybe creating two tests and having some duplication is better? WDYT?

Reply:
done